libs/core/src/lib/select/select-key-manager.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable } from '@angular/core';
+import { takeUntil } from 'rxjs/operators';
+import { ActiveDescendantKeyManager } from '@angular/cdk/a11y';
+import { KeyUtil } from '../utils/functions';
+import { UP_ARROW, DOWN_ARROW, ENTER, SPACE, HOME, END, hasModifierKey } from '@angular/cdk/keycodes';
+
+import { OptionComponent } from './option/option.component';
+
+
+@Injectable()
+export class SelectKeyManagerService {
+
+    _component: any;
+    _keyManager: ActiveDescendantKeyManager<OptionComponent>;
+
+    /**
+    * Sets up a key manager to listen to keyboard events on the overlay panel.
+    * @hidden
+    */
+    _initKeyManager(_component: any): void {
+        this._keyManager = new ActiveDescendantKeyManager<OptionComponent>(this._component._options)
```

Comment:
It would be great to have some interface, instead of `OptionComponent`, just to not restrict this service to select component